lib/decode.c

```diff
@@ -916,3 +916,19 @@ quirc_decode_error_t quirc_decode(const struct quirc_code *code,
 
 	return QUIRC_SUCCESS;
 }
+
+void quirc_flip(struct quirc_code *code)
+{
+    struct quirc_code flipped;
+    memset(&flipped, 0, sizeof(flipped));
+    int offset = 0;
+    for (int y = 0, sx = 0; y < code->size; y++, sx++) {
+        for (int x = 0, sy = 0; x < code->size; x++, sy++) {
+            if (grid_bit(code, sx, sy)) {
```

Comment:
nit: My understanding here is that `y == sx` and `x == sy` always hold. I guess the intent is to avoid `grid_bit(code, y, x)` because `x` and `y` would be in the inverse order of the definition:
```c
static inline int grid_bit(const struct quirc_code *code, int x, int y)
```

Personally, in the context of "flipping", I wouldn't mind this inversion. It would actually explicit the flipping if anything.